src/components/Table/DataTable/compenents/el-data-table/utils/select-strategy.js

```diff
@@ -83,33 +91,42 @@ class StrategyPersistSelection extends StrategyAbstract {
   toggleRowSelection(row, isSelected) {
     const { id, selected } = this.elDataTable
     const foundIndex = selected.findIndex(r => r[id] === row[id])
+
     if (typeof isSelected === 'undefined') {
       isSelected = foundIndex <= -1
     }
+
     if (isSelected && foundIndex === -1) {
       selected.push(row)
     } else if (!isSelected && foundIndex > -1) {
       selected.splice(foundIndex, 1)
     }
+
     this.elDataTable.$emit('toggle-row-selection', isSelected, row)
     this.updateElTableSelection()
   }
+
   clearSelection() {
     this.elDataTable.selected = []
     this.updateElTableSelection()
   }
+
   /**
    * 将selected状态同步到el-table中
    */
   updateElTableSelection() {
     const { data, id, selected } = this.elDataTable
+
     // 历史勾选的行已经不在当前页了，所以要将当前页的行数据和selected合并
     const mergeData = _.uniqWith([...data, ...selected], _.isEqual)
+
     mergeData.forEach(r => {
       const isSelected = !!selected.find(r2 => r[id] === r2[id])
+
       if (!this.elTable) {
         return
       }
+
       this.elTable.toggleRowSelection(r, isSelected)
     })
   }
```

Comment:
This code appears to define a set of methods within an interface. The code doesn't seem like it's optimized or contains issues that need addressing due to being so short and minimal. It could benefit from better documentation or comments explaining why certain methods behave as they do, but this is not necessary for understanding the problem you gave me.

However, there is potentially an opportunity for improving readability with more explicit naming conventions and possibly adding more comments, especially when dealing with new types of data structures such as `selected` which seems to represent both previously selected items and newly added ones.